api/sendfild/pb/sendfil.proto

```diff
@@ -0,0 +1,103 @@
+syntax = "proto3";
+package api.sendfild.pb;
+option go_package = "github.com/textileio/textile/v2/api/sendfild/pb";
+
+import "google/protobuf/timestamp.proto";
+
+enum MessageState {
+  MESSAGE_STATE_UNSPECIFIED = 0;
+  MESSAGE_STATE_PENDING = 1;
+  MESSAGE_STATE_ACTIVE = 2;
+  MESSAGE_STATE_FAILED = 3;
```

Comment:
This is a final state that can't be recovered from. The message is permanently marked as failed.

Reply:
The `Txn.failure_msg` will explain why.